src/cockpit/channel.py

```diff
@@ -433,31 +433,38 @@ class AsyncChannel(Channel):
     """
 
     # Receive-side flow control: intermix pings and data in the queue and reply
-    # to pings as we dequeue them.  This is a buffer: since we need to handle
-    # do_data() without blocking, we have no choice.
-    receive_queue = None
+    # to pings as we dequeue them.  EOF is None.  This is a buffer: since we
+    # need to handle do_data() without blocking, we have no choice.
+    receive_queue: 'asyncio.Queue[bytes | JsonObject | None]'
 
     # Send-side flow control
     write_waiter = None
 
-    async def run(self, options):
+    async def run(self, options: JsonObject) -> None:
         raise NotImplementedError
 
-    async def run_wrapper(self, options):
+    async def run_wrapper(self, options: JsonObject) -> None:
         try:
             await self.run(options)
             self.close()
         except ChannelError as exc:
             self.close(exc.attrs)
 
-    async def read(self):
+    async def read(self) -> 'bytes | None':
+        # Three possibilities for what we'll find:
+        #  - None (EOF) → return b''
```

Comment:
This is a bit confusing as that's not what the code below does -- it returns `None`, not b''. Given that you changed everything to `None`, is this a leftover from a previous version?